src/twitter/Timeline.js

```diff
@@ -97,40 +132,46 @@ class Timeline {
       obj = this._hashtags[htag]
     }
     else {
-      obj = {hashtag, room: [] }
+      obj = {hashtag, room: new Set() }
       if(opts.is_new) {
         this._newtags.add("#"+hashtag);
       }
     }
-    if(!obj.room.includes(room_id)) {
-      obj.room.push(room_id);
-    }
+    obj.room.add(room_id);
     if(htag !== -1) {
-      obj = this._hashtags[htag] = obj;
+      this._hashtags[htag] = obj;
     }
     else {
       this._hashtags.push(obj);
     }
     log.info("Added Hashtag: %s", hashtag);
+    return true;
   }
 
   /**
    * Add a Twitters user's timeline to the timeline processor. Tweets will be
    * automatically send to the given room.
    *
+   * HTM* - Use a single account to send tweets, avoiding large numbers of join
+   * events. This is enabled automatically unless disabled via config
+   *
    * @param  {string} twitter_id The twitter ID of a timeline.
    * @param  {string} room_id The room_id to insert tweets into.
    * @param  {object} opts Options
    * @param  {boolean} opts.is_new Is the room 'new', and we shouldn't do a full poll.
    * @param  {boolean} opts.exclude_replies Should we not fetch replies.
-   */
+   * @param  {boolean} opts.high_traffic_mode Enable high traffic mode on the timeline*.
+   * @return {boolean} was the hashtag added/changed
+  */
   add_timeline (twitter_id, room_id, opts) {
-    var tline = this._find_timeline(twitter_id);
-    var obj;
-
     if (this.config.timelines.enable === false) {
-      return;
+      return false;
+    }
+    if (!Util.isRoomId(room_id)) {
```

Comment:
Hmm, I was probably being paranoid here. I don't think any routes to add_timeline accept pure roomIds without checking them.